doc/notes/11455_deployments/11455_export_depl.dnt.adoc

```diff
@@ -0,0 +1,227 @@
+
+Added templates:
+
+* t.scenario_list_begin.wasl
+* t.scenario_list_end.wasl
+
+```
+$START_SCENARIOS
+<project.name>,<project.version>,<project.name>::<activity.name>,<routine.number>,<project.name>::<activity.name>Scenario<routine.number>.scn
+$END_SCENARIOS
+```
+
+==== 5.2.2.7 Hand coded file pointer
+
+Added templates:
+
+* t.hcf.wasl
+
+```
+$START_
+<project.keyletters>.hcf
+$END_
+```
+
+==== 5.2.2.7.1 <project.keyletters>.hcf
+This file is output with a list of non-sequenced hand written terminators.  To support gathering the list of such files the code in genfile::render_wasl is refactored into a new operation, domain.render_terminators_handcoded().  This new operation is also called within the project.render_wasl() operation.
+
+==== 5.2.2.8 Domains in build set list
+Waiting on required around this one.  To support it a new operation is added to project, render_domains_in_build_set().  For now the two templates below are called to produce the required entry.
+
+Added template:
+
+* t.domains_in_build_set_begin.wasl
+* t.domains_in_build_set_end.wasl
+
+```
+$START_DOMAINS_IN_BUILD_SET
+$END_DOMAINS_IN_BUILD_SET
+```
+
+==== 5.2.2.9 Bridge Data Types
+A reference to a file containing a breakdown of the parameter types of each local terminator.
+
+These will map to non-sequenced Deployment scoped Terminators in xtUML.  Code that gathers this in the domain case already exists.  That code is refactored into a new operation on domain, render_bridges()[[render_bridges]].  This operation is called in genfile.render_wasl() as well as in project.render_wasl().
+
+Added templates:
+
+* t.bridge_data_types_start.wasl
+* t.bridge_data_types_end.wasl
+
+```
+$START_BRIDGE_DATA_TYPES
+<project.keyletters>.brp
+$END_BRIDGE_DATA_TYPES
+```
+
+==== 5.2.2.9.1 <project.keyletters>.brp
+A new operation is added to domain, render_bridge_list().  This operation renders each routine under the domain, and outputs the list into the file.
+
+==== 5.2.2.10 Bridges
+The domain.render_bridges() operation that was refactored in <<render_bridges>>, takes care of creating the required .br file for each non sequenced terminator.  The operation is also written to collect the list of these terminators.  This list is passed to a new operation on the project class, render_bridges.  This operation simply outputs the given list between the templates below.
+
+Added templates:
+
+* t.bridges_begin.wasl
+* t.bridges_end.wasl
+
+```
+$START_BRIDGES
+<domain.name>,<project.version>,<domain.name>_<routine.number>_<activity.name>.br,<contains output ? Y : N>
```

Comment:
I am not sure if there is where I saw it, but there is a heavy leaning on string arithmetic in places where we could populate a template (for comma separate value output).  This is not critical, but helpful for maintenance going forward.

Reply:
I agree here, but will leave that for the next set of work.